borsh-ts/index.ts

```diff
@@ -0,0 +1,217 @@
+import BN from 'bn.js';
+
+const INITIAL_LENGTH = 1024;
+
+export type Schema = Map<Function, any>;
+
+/// Binary encoder.
+export class BinaryWriter {
+    buf: Buffer;
+    length: number;
+
+    public constructor() {
+        this.buf = Buffer.alloc(INITIAL_LENGTH);
+        this.length = 0;
+    }
+
+    maybe_resize() {
```

Comment:
it's weird to use `snake_case` in JS code. Especially given there is other stuff not following same convention, e.g `serializeField`